app/lib/task/backend.dart

```diff
@@ -686,13 +686,14 @@ class TaskBackend {
       final key = PackageState.createKey(_db, runtimeVersion, package);
       final state = await tx.lookupOrNull<PackageState>(key);
       if (state == null || state.versions![version] == null) {
-        throw NotFoundException.resource('$package/$version');
+        throw TaskAbortedException(
+            '$package/$version is no longer selected for analysis.');
       }
       final versionState = state.versions![version]!;
 
       // Check the secret token
       if (!versionState.isAuthorized(_extractBearerToken(request))) {
-        throw AuthenticationException.authenticationRequired();
+        throw TaskAbortedException('Secret token is no longer accepted.');
```

Comment:
How do we know that this isn't because the secret is malformed?

Reply:
Does it matter to the worker if it is aborted because the secret was malformed?
I think it is only the server that may be interested in malformed tokens, as it could report them as potential attack attempt... or is there any other use case for it?

Reply:
Well,

 * On the **server side**, a _malformed token_ is an indicator of:
    * (A.1) Unknown internal bug (producing garbled payload, etc),
    * (B.1) Worker reporting back a canceled job, or,
    * (C) Being under attack (in which case we wish to reveal as little as possible, to an attacker).
 * On the **worker side**, a _malformed token_ is an indicator of:
    * (A.2) Unknown internal bug (receiving garbled payload, decoding payload incorrectly, etc), or,
    * (B.2) Worker reporting back a canceled job.

(A) is something we shout into a log, otherwise we'll never find out.
(B) is harmless, should be raised as an info or warning in logs.
(C) we should return `401`.

If (A) is a silent error, then why would we even bother looking at worker logs?

Reply:
Let's spell out the cases:

If it is missing the bearer token, or it has an empty one, then it is a `401`.

If the server has no token to compare with, then it is likely that the task was aborted and we could return `400`+`TaskAborted`. But it may have been an attack attempt to inject something for a non-analyzed version. I think we can't genuinely decide here, but erring for the task being aborted seems to be okay.

After both is present, IIRC we don't have internal structure for these tokens, and do only string comparison. If it is non-matching, we could probably return `403` and shout in the logs...

Is these all what we are looking for here?

Reply:
I'm not sure :see_no_evil: 

Perhaps we should consider introducing a `runId` or similar thing, I'm not sure.

Reply:
Updated with tracking the aborted tokens. PTAL!